### PR TITLE
Create first draft of reusable workflows

### PR DESCRIPTION
**Pull request by lucasmellos**
Supports the following actions:
- SonarQube code coverage
- JavaScript, Ruby, and .NET unit testing
- helm deploy
- Terragrunt based apply
- Docker Swarm deploy based on Portainer
- Step Approval
- GPG decryption

Samples on how to use these workflows will be added in the future

### CODE REVIEW

File: .github/actions/test-ruby/action.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        bundler-cache: true
+        version: ${{ inputs.RUBY_VERSION }}
+
+    - name: Run Tests
+      run: |
+        cp .env.example .env
+        bundle install --path ./bundle --without production,development > /dev/null
+        bundle exec rake db:test:prepare > /dev/null
+        bundle exec rake core:db:test:prepare > /dev/null
+        bundle exec rake messaging:db:test:prepare > /dev/null
```
Comment (Null-Is-Null) [inline .github/actions/test-ruby/action.yml:25]: For Rails projects other than prime-rails that don't have a core or messaging database, will these lines break the pipeline?

File: .github/actions/unit-test-dotnet/action.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    required: false
+    default: "3.1.402"
+
+runs:
+  using: "composite"
+  steps:
+    - name: Setup .NET Core
+      uses: actions/setup-dotnet@v1
+      with:
+        dotnet-version: ${{ inputs.DOTNET-VERSION }}
```
Comment (Null-Is-Null) [inline .github/actions/unit-test-dotnet/action.yml:15]: Should this be `inputs.DOTNET_VERSION`? Not sure if having a dash here will cause issues. Other inputs in these artifacts consistently use underscore instead.

File: .github/workflows/docker-bp.yml [hunk 3]
```diff
@@ -1,147 +1,109 @@
...
-        EXECUTIONLOG="${EXECUTIONLOG//$'\r'/'%0D'}"
-        echo "::set-output name=container-logs::$EXECUTIONLOG"
-
-    - name: Update Pull Request
-      uses: actions/github-script@v6
-      if: github.event_name == 'pull_request' && github.event.inputs.PR_MESSAGE == true
-      env:
-        LOGPRINT: "${{ steps.test.outputs.container-logs }}"
+        repository: signalwire/actions-template
+        ref: dev
```
Comment (Null-Is-Null) [inline .github/workflows/docker-bp.yml:78]: Will this ref still be correct after merging this PR?

**Review by Null-Is-Null — CHANGES_REQUESTED** [review]
Looks pretty good, just had a few comments. In addition to my last comment about `ref: dev`, it looks like `@dev` is referenced in the sample workflows which may have to be changed as well.

**Review by Null-Is-Null — APPROVED** [review]
Looks good, approving. Does my comment on `.github/actions/test-ruby/action.yml` raise any concerns though?

**Comment by lucasmellos** [comment]
We'll have to test it, I spent the afternoon working on conference-manager

**Pull request merged**